tools/offwaketime.py

```diff
@@ -288,15 +292,20 @@ def signal_ignore(signal, frame):
 has_enomem = False
 counts = b.get_table("counts")
 stack_traces = b.get_table("stack_traces")
+need_delimiter = args.delimited and not (args.kernel_stacks_only or
+                                         args.user_stacks_only)
 for k, v in sorted(counts.items(), key=lambda counts: counts[1].value):
     # handle get_stackid errors
-    # check for an ENOMEM error
-    if k.w_k_stack_id == -errno.ENOMEM or \
-       k.t_k_stack_id == -errno.ENOMEM or \
-       k.w_u_stack_id == -errno.ENOMEM or \
-       k.t_u_stack_id == -errno.ENOMEM:
-        missing_stacks += 1
-        continue
+    if not args.user_stacks_only:
+        missing_stacks += int(stack_id_err(k.w_k_stack_id))
+        missing_stacks += int(stack_id_err(k.t_k_stack_id))
+        has_enomem = has_enomem or (k.w_k_stack_id == -errno.ENOMEM) or \
+                     (k.t_k_stack_id == -errno.ENOMEM)
+    if not args.kernel_stacks_only:
+        missing_stacks += int(stack_id_err(k.w_u_stack_id))
+        missing_stacks += int(stack_id_err(k.t_u_stack_id))
+        has_enomem = has_enomem or (k.w_u_stack_id == -errno.ENOMEM) or \
+                     (k.t_u_stack_id == -errno.ENOMEM)
 
     waker_user_stack = [] if k.w_u_stack_id < 1 else \
         reversed(list(stack_traces.walk(k.w_u_stack_id))[1:])
```

Comment:
@brendangregg Do you still remember what's reason for doing the `[1:]` (and we don't do it in non-folded case)?